src/WrappedVault.sol

```diff
@@ -349,7 +362,7 @@ contract WrappedVault is Owned, ERC20, IWrappedVault {
 
         // If there are no stakers we just change the last update time, the rewards for intervals without stakers are not accumulated
 
-        uint256 elapsedWAD = elapsed * 1e18;
+        uint256 elapsedWAD = elapsed * 10 ** (decimalOffset);
```

Comment:
```suggestion
        uint256 elapsedWAD = elapsed * 10 ** (ERC20(address(vault)).decimals() + decimalOffset);
```

The `decimalOffset` will be 0 if the reward token and the asset token have the same decimals, and there is no decimal offset. In that case, the accumulated rewards will often be zero as `totalSupply` would be larger than `elapsed * rate`.

The risk here is to overflow rewardsPerTokenOut.accumulated.

`elapsed` was originally scaled up by 1e18. If we scale it up by 1e(18+delta): 
- There is no effect for reward/vault combinations that are 18/18.
- For reward/vault combinations that are 18/6 there is no change wrt to the original code.
- For reward/vault combinations that are 6/18. elapsed is scaled up an additional 1e12, but rate is scaled down by 1e12, so we stay neutral.